cloudinary-test-common/src/main/java/com/cloudinary/test/AbstractStructuredMetadataTest.java

```diff
@@ -0,0 +1,241 @@
+
+        Date legalValue = new Date(min.getTime() + 36 * 60 * 60 * 1000);
+        Date illegalValue = new Date(max.getTime() + 36 * 60 * 60 * 1000);
+
+        DateMetadataField dateMetadataField = new DateMetadataField();
+        dateMetadataField.setLabel("Start date" + new Date().getTime());
+
+        List<MetadataValidation> rules = new ArrayList<MetadataValidation>();
+        rules.add(new MetadataValidation.DateGreaterThan(min));
+        rules.add(new MetadataValidation.DateLessThan(max));
+        dateMetadataField.setValidation(new MetadataValidation.AndValidator(rules));
+
+        String message = null;
+        ApiResponse res = null;
+        try {
+            // should fail
+            dateMetadataField.setDefaultValue(illegalValue);
+            res = api.addMetadataField(dateMetadataField);
+            // this line should not be reached if all is working well, but when it's not we still want to clean it up:
+            metadataFieldExternalIds.add(res.get("external_id").toString());
+        } catch (BadRequest e) {
+            message = e.getMessage();
+        }
+
+        assertEquals(message, "default_value is invalid");
+
+        // should work:
+        dateMetadataField.setDefaultValue(legalValue);
+        res = api.addMetadataField(dateMetadataField);
+        metadataFieldExternalIds.add(res.get("external_id").toString());
+    }
+
+    @Test
+    public void testListFields() throws Exception {
+        StringMetadataField stringField = newFieldInstance("testListFields");
+        addFieldToAccount(stringField);
+
+        ApiResponse result = cloudinary.api().listMetadataFields();
+        assertNotNull(result);
+    }
+
+    @Test
+    public void testGetMetadata() throws Exception {
+        ApiResponse fieldResult = addFieldToAccount(newFieldInstance("testGetMetadata"));
+        ApiResponse result = api.metadataFieldByFieldId(fieldResult.get("external_id").toString());
+        assertNotNull(result);
+        assertEquals(fieldResult.get("label"), result.get("label"));
+    }
+
+    @Test
+    public void testUpdateField() throws Exception {
+        ApiResponse fieldResult = addFieldToAccount(newFieldInstance("testUpdateField"));
+        assertNotEquals("new_def", fieldResult.get("default_value"));
+        StringMetadataField field = new StringMetadataField();
+        field.setDefaultValue("new_def");
+        ApiResponse result = api.updateMetadataField(fieldResult.get("external_id").toString(), field);
+        assertNotNull(result);
+        assertEquals("new_def", result.get("default_value"));
+    }
+
+    @Test
+    public void testDeleteField() throws Exception {
+        ApiResponse fieldResult = addFieldToAccount(newFieldInstance("testDeleteField"));
+        ApiResponse result = api.deleteMetadataField(fieldResult.get("external_id").toString());
+        assertNotNull(result);
+        assertEquals("ok", result.get("message"));
```

Comment:
Check that getMetadataField returns null

Reply:
Why? Java tests are slow as it is, and subsequent calls are 100% server tests with no implications on the SDK whatsoever. Since we do assert the server ACK, checking again will turn this into a pure backend regression test...